Add description of output files to scpca-nf instructions (#124)
AlexsLemonade/scpca-nf: external-data-instructions.md

<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*

> - [How to use `scpca-nf` as an external user](#how-to-use-scpca-nf-as-an-external-user)

Reviewer: Funny how my toc generator decided that it should remove this top level header and yours added it back. No required change, just interesting... I'll try to make sure not to commit such changes if mine does that next time.

All files in that folder will be prefixed by library ID, with the following suffixes:  `_unfiltered.rds`, `_filtered.rds`, `_metadata.json`, and `_qc.html`. 
The `_unfiltered.rds` and `_filtered.rds` files contain the quantified gene expression data as a [`SingleCellExperiment` object](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html).
For more information on the contents of these files, see the [ScPCA portal docs section on single cell gene expression file contents](https://scpca.readthedocs.io/en/latest/sce_file_contents.html).

See below for the expected structure of the `publish` folder: 

```
publish
└── SCPCS999991
    ├── SCPCL999991_filtered.rds
    ├── SCPCL999991_metadata.json
    ├── SCPCL999991_qc.html
    └── SCPCL999991_unfiltered.rds
```

If bulk libraries were processed, a `bulk_quant.tsv` and `bulk_metadata.tsv` summarizing the counts data and metadata across all libraries will also be present in the `publish` directory. 

The `internal` folder will contain intermediate files that are produced by individual steps of the workflow, including mapping with `salmon` and quantification with `alevin-fry`. 

> The `rad` folder contains the output from running [`salmon alevin`](https://salmon.readthedocs.io/en/latest/alevin.html#alevin) with the `--rad` flag, while the `af` folder contains the outputs from [`alevin-fry`](https://alevin-fry.readthedocs.io/en/latest/index.html). 
> These are created in part to allow restarting the workflow from internal checkpoints (in particular so the initial `salmon` mapping does not need to be repeated).
> These also may contain log files and other outputs that can be useful for troubleshooting or alternative analyses.
> 
> All files pertaining to a specific library will be nested within a folder labeled with the library ID.
> Additionally, for each run, all files related to that run will be inside a folder labeled with the run ID followed by the type of run (i.e. `rna` or `features` for CITE-seq) and nested within the library ID folder.
> 
> If bulk libraries are processed, there will be an additional `salmon` folder that contains the output from running [`salmon`](https://salmon.readthedocs.io/en/latest/file_formats.html#fileformats) on each library processed. 

Reviewer: I moved the comment about the extra folder in bulk up, so that all the potential folders are listed together. 
I also moved the why to the second sentence, before listing the folders, with some minor rewording.

```suggestion
The contents of this folder are used to allow restarting the workflow from internal checkpoints (in particular so the initial read mapping does not need to be repeated), and may contain log files and other outputs useful for troubleshooting or alternative analysis.

The `rad` folder contains the output from running [`salmon alevin`](https://salmon.readthedocs.io/en/latest/alevin.html) with the `--rad` flag, while the `af` folder contains the outputs from [`alevin-fry`](https://alevin-fry.readthedocs.io/en/latest/index.html). 
If bulk libraries are processed, there will be an additional `salmon` folder that contains the output from running [`salmon quant`](https://salmon.readthedocs.io/en/latest/file_formats.html) on each library processed. 

All files pertaining to a specific library will be nested within a folder labeled with the library ID.
Additionally, for each run, all files related to that run will be inside a folder labeled with the run ID followed by the type of run (i.e. `rna` or `features` for CITE-seq) and nested within the library ID folder.
```

## Special considerations for using `scpca-nf` with spatial transcriptomics libraries 

Instructions on creating your own spaceranger docker image

## Output files 

Upon completion of the `scpca-nf` workflow, the results will be published to the specified `outdir`. 
Within the `outdir`, two folders will be present, `publish` and `internal`. 

The `publish` folder will contain the final output files produced by the workflow and the files that are typically available for download on the ScPCA portal. 

Within the `publish` folder, all files pertaining to a specific sample will be nested within a folder labeled with the sample ID.

> An `unfiltered.rds`, `filtered.rds`, `metadata.json`, and QC report (`qc.html`) will be present for each library. 

Reviewer: Rephrasing a bit, trying to be a bit more clear on the file names. Not sure I am succeeding.
```suggestion
All files in that folder will be prefixed by library ID, with the following suffixes:  `_unfiltered.rds`, `_filtered.rds`, `_metadata.json`, and `_qc.html`. 
```

> The `unfiltered.rds` and `filtered.rds` files will contain the quantified gene expression data as a [`SingleCellExperiment` object](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html).

Reviewer: Minor wording & matching changes above.
```suggestion
The `)unfiltered.rds` and `_filtered.rds` files contain the quantified gene expression data as a [`SingleCellExperiment` object](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html).
```

> For more information on the contents of these files, see the [ScPCA portal docs section on single cell gene expression file contents.](https://scpca.readthedocs.io/en/latest/sce_file_contents.html)

Reviewer: Period location
```suggestion
For more information on the contents of these files, see the [ScPCA portal docs section on single cell gene expression file contents](https://scpca.readthedocs.io/en/latest/sce_file_contents.html).
```

See below for the expected structure of the `publish` folder: 

```

> └── SCPCS999991

Reviewer: I think making the relative root explicit in the tree is probably worth doing.
~~~suggestion
```
publish
└── SCPCS999991
~~~

    ├── SCPCL999991_filtered.rds
    ├── SCPCL999991_metadata.json
    ├── SCPCL999991_qc.html
    └── SCPCL999991_unfiltered.rds
```

> Additionally, if bulk libraries are processed, a `bulk_quant.tsv` and `bulk_metadata.tsv` summarizing the counts data and metadata across all libraries will be present in the `publish` directory. 

Reviewer: Minor wording
```suggestion
If bulk libraries were processed, a `bulk_quant.tsv` and `bulk_metadata.tsv` summarizing the counts data and metadata across all libraries will also be present in the `publish` directory. 
```

> The `internal` folder will contain the intermediate files that are produced throughout the workflow. 

Reviewer: I wonder if we need this level of detail about the internal folder. We might be able to get away with not showing the tree, and instead simply giving a little more background of what is there and why. 

```suggestion
The `internal` folder will contain intermediate files that are produced by individual steps of the workflow, including mapping with `salmon` and quantification with `alevin-fry`. 
These are created in part to allow restarting the workflow from internal checkpoints (in particular so the initial `salmon` mapping does not need to be repeated).
These also may contain log files and other outputs that can be useful for troubleshooting or alternative analyses.
```

All files pertaining to a specific library will be nested within a folder labeled with the library ID.
Additionally, for each run, all files related to that run will be inside a folder labeled with the run ID followed by the type of run (i.e. `rna` or `features` for CITE-seq) and nested within the library ID folder.

> The `af` folder contains the output from running [`salmon alevin`](https://salmon.readthedocs.io/en/latest/alevin.html#alevin) with the `--rad` flag, while the `rad` folder contains the outputs from [`alevin-fry`](https://alevin-fry.readthedocs.io/en/latest/index.html). 

Reviewer: I think these were reversed. I would also put this sentence before the discussion of libraries & runs, to follow the nesting structure of the output. 

```suggestion
The `rad` folder contains the output from running [`salmon alevin`](https://salmon.readthedocs.io/en/latest/alevin.html#alevin) with the `--rad` flag, while the `af` folder contains the outputs from [`alevin-fry`](https://alevin-fry.readthedocs.io/en/latest/index.html). 
```

> See below for the expected structure of the `internal` folder after running the single-cell/single-nuclei workflow: 

Reviewer: I think we can get away with not including this at all. Part of my thought is that if any internal outputs change, we would have to update this, which is a pain, but more is that if somebody does want to dive into the internals, they ought to be referring to the salmon/alevin fry documentation, which you have already linked to.
